### PR TITLE
Fix build issues with latest cython

### PR DESCRIPTION
**Pull request by chaithyagr**

### CODE REVIEW

File: pynfft/nfft.pyx [hunk 1]
```diff
@@ -34,7 +34,7 @@ np.import_array()
 fftw_init_threads()
 
 # register cleanup callbacks
-cdef void _cleanup():
+cdef void _cleanup() noexcept:
```
Comment (ghisvail) [inline pynfft/nfft.pyx:37]: Why is this needed?

File: setup.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def get_cython_extensions():
             **common_extension_args
             )
         )
-    return cythonize(ext_modules)
+    return cythonize(ext_modules, language_level=2)
```
Comment (ghisvail) [inline setup.py:89]: Why is this needed?

**Review by ghisvail — COMMENTED** [review]
Needs more details about the changes. So far, this PR gives no context about the build issues and what's needed to fix them. Thanks.

**Comment by chaithyagr** [comment]
Both the above changes are just cause the coding styles have changed with cython 3.0 and later. This just ensures we are compatible. There was really not much other than "to be compatible with Cython 3.0+"
